### DISCUSSION TITLE
DFT energy and predicted energy off by 4eV

### DISCUSSION THREAD

**Discussion by MathildeAnd** [deepmodeling/deepmd-kit#3804] [Q&A]
Hello everyone, 

I'm doing ML from systems composed by Fe+H2O and FeO+H2 at 2000K, 3000K and 4000K and I trained the potential for each separate temperature. I encounter a problem when testing the model : the predicted energy and the DFT energy are off by approximately 4 eV. With my training data set composed by AIMD by VASP, I used LDA+U with a J parameter of 1 and a U of 5 so the gap in the energies seems to correspond to the LDA+U parameter. However I don't understand how to fix this problem, do you have any suggestions ? 
I attached my input and the figure I obtain for the predicted energy against DFT energy 
<img width="475" alt="Capture d’écran 2024-05-22 à 11 55 54" src="https://github.com/deepmodeling/deepmd-kit/assets/170516211/a5be37f7-cdc0-472d-8df1-4aa5bd06f57e">
[input_MyPotential.json](https://github.com/deepmodeling/deepmd-kit/files/15401498/input_MyPotential.json)

**Comment by njzjz** [comment]
I see that your training data contains systems with different number of atoms. Do you use the same LDA+U parameter for these systems? Do all of these systems have the same bias?

**Reply by MathildeAnd** [reply]
Yes I use the same LDA+U parameter for all the systems, I just adjust it to the number of iron in my system.

**Reply by njzjz** [reply]
Do you see the consistent error in `lcurve.out`?

**Reply by MathildeAnd** [reply]
<img width="683" alt="Capture d’écran 2024-05-24 à 09 51 25" src="https://github.com/deepmodeling/deepmd-kit/assets/170516211/8b5e5754-3942-4cbb-8e32-ac3d8b10acbe">

Here is the lcurve.out file for 2000K system.

**Reply by njzjz** [reply]
It looks to me like some points have inconsistent 1e-4 `rmse_e_trn`, which is much smaller than the bias shown in the figure.

**Reply by njzjz** [reply]
Could you plot all the points in the training data instead of some of them?

**Reply by MathildeAnd** [reply]
You mean all of the 10.000 points ?
Could it be an overfitting problem ?

**Comment by wanghan-iapcm** [comment]
So you have trained three DP models,  for 2000K, 3000K and 4000K, respectively. The labeling method is LDA+U with J=1 and U=5

What was the model for the test plot? How was the test dataset constructed?

**Reply by MathildeAnd** [reply]
The test plot shows the DP model for 2000K but it does the same for the other temperatures. 
I choose to test the model with the simulation in which I had the most frames. So it is composed of one simulation only.

**Reply by wanghan-iapcm** [reply]
how many atoms do you have in your configurations? could you please show the command line output of the `dp test`?

**Reply by MathildeAnd** [reply]
For Fe+H2O I have two different systems :
1) with 100 iron and 40 H2O so 220 atoms
2) with 100 iron and 70 H2O so 310 atoms 
For FeO+H2 I have also two systems :
1) with 256 FeO (half-half) and 50 H2 so 356 atoms
2) with 156 FeO (half-half) and 78 H2 so 312 atoms

I'm not sure what you mean by the command line output of the dp test but the final output command was this : 
DEEPMD INFO    Energy RMSE        : 4.210666e+00 eV
DEEPMD INFO    Energy RMSE/Natoms : 1.913939e-02 eV
DEEPMD INFO    Force  RMSE        : 2.433533e-01 eV/A
DEEPMD INFO    Virial RMSE        : 6.228288e+00 eV
DEEPMD INFO    Virial RMSE/Natoms : 2.831040e-02 eV

**Reply by wanghan-iapcm** [reply]
the std of your energy label per atom is approximately 8 / 300 = 26 meV/atom, which is relatively small. 
the energy training error is 19 meV/atom, mainly contributed by the "constant shift"

to fix the issue it is recommended to refine your model with reduced force prefactors, e.g. 1-> 1 and increased energy prefactors e.g. 1->10 and a reduced start learning rate, e.g. 10 times smaller than from scratch training

you may use the `init-model` mode to load model parameters and start a refinement, see https://docs.deepmodeling.com/projects/deepmd/en/r2/train/training-advanced.html#options-and-environment-variables

**Reply by MathildeAnd** [reply]
Ok thank you for your suggestions, I tried to tweak this parameters but I still end up with a residual error on the forces of 1 eV...

**Reply by wanghan-iapcm** [reply]
if you find you cannot simultaneously improve the energy and force accuracies, you may check the quality of your data following https://docs.deepmodeling.com/projects/deepmd/en/r2/troubleshooting/precision.html
